smartsim/settings/batch_settings.py

```diff
@@ -140,6 +149,16 @@ def env_vars(self) -> StringArgument:
     @env_vars.setter
     def env_vars(self, value: t.Dict[str, str | None]) -> None:
         """Set the environment variables."""
+
+        if not (
+            isinstance(value, t.Mapping)
```

Comment:
Minor nit: Here we are checking for a `Mapping`, but the type hint is asking specifically for a `dict`. Can you change the type hint to reflect this?

Reply:
changed